server/src/main/java/io/crate/execution/ddl/tables/AlterTableOperation.java

```diff
@@ -289,6 +295,32 @@ static void validateReadOnlyIndexForResize(IndexMetadata indexMetadata) {
         }
     }
 
+    private static void validateForPublishedTables(BoundAlterTable alterTable,
+                                                   ClusterState currentState,
+                                                   IndexScopedSettings indexScopedSettings) {
+        PublicationsMetadata publicationsMetadata = currentState.metadata().custom(PublicationsMetadata.TYPE);
+        if (publicationsMetadata != null) {
+            var publications = publicationsMetadata.publications();
+            for (var publication : publications.values()) {
+                for (var table : publication.tables()) {
+                    if (table.equals(alterTable.table().ident())) {
+                        for (var key : alterTable.tableParameter().settings().keySet()) {
+                            if (!indexScopedSettings.isDynamicSetting(key)) {
+                                throw new IllegalArgumentException(
+                                    String.format(
+                                        "Setting [%s] cannot be applied to table '%s' published for logical replication",
```

Comment:
```suggestion
                                        "Setting [%s] cannot be applied to table '%s' because it is included in a logical replication publication '%s'",
```